### PR TITLE
Remove dependency on ppx_bin_prot and ppx_deriving_yojson

### PR DESCRIPTION
**Pull request by bbc2**
There were already deprecated.  This also fixes the CI and the changelog.

### CODE REVIEW

File: lib/cvc.ml [hunk 1]
```diff
@@ -237,7 +237,7 @@ struct
       n: Derivable.Z.t;
       e: Derivable.Z.t;
     }
-    [@@deriving ord,eq,yojson,eq,show,bin_io]
+    [@@deriving ord,eq,eq,show]
```
Comment (ian-barnes) [inline lib/cvc.ml]: Is there a reason `eq` is listed twice? (Here and also below.)

Comment (bbc2) [reply lib/cvc.ml]: Fixed. Thanks!

**Comment by eXenon** [comment]
Looks good to me.

**Comment by bbc2** [comment]
I thought you would merge since you're assigned but I'll do it since everybody approves.

**Pull request merged**